Bug: Visiting /v/posts crashes the backend

## Expected Behavior
Visiting /v/posts should either display a page of unfiltered results or display some error (400 bad request?). The backend should continue running normally.

## Actual Behavior
The backend crashes. Stack trace:
```
panic: runtime error: index out of range [0] with length 0

goroutine 74 [running]:
github.com/OpenEugene/openboard/back/internal/postsvc/internal/postdb.(*PostDB).findPosts(0xc00000e7a0, 0xb4a8e0, 0xc0002c7dd0, 0xc0002df310, 0xc0002ec0c0, 0x0, 0x0)
	/home/appins/openboard/back/internal/postsvc/internal/postdb/qrypost.go:121 +0xa26
github.com/OpenEugene/openboard/back/internal/postsvc/internal/postdb.(*PostDB).FndPosts(0xc00000e7a0, 0xb4a8e0, 0xc0002c7dd0, 0xc0002df310, 0xa3e300, 0x0, 0xb4ce80)
	/home/appins/openboard/back/internal/postsvc/internal/postdb/postdb.go:63 +0x6c
github.com/OpenEugene/openboard/back/internal/postsvc.(*PostSvc).FndPosts(0xc000059b80, 0xb4a8e0, 0xc0002c7dd0, 0xc0002df310, 0xc000059b80, 0xc0002c7dd0, 0xc000053a20)
	/home/appins/openboard/back/internal/postsvc/postsvc.go:60 +0x51
github.com/OpenEugene/openboard/back/internal/pb._Post_FndPosts_Handler(0xa3e300, 0xc000059b80, 0xb4a8e0, 0xc0002c7dd0, 0xc0002df2c0, 0x0, 0xb4a8e0, 0xc0002c7dd0, 0x0, 0x0)
	/home/appins/openboard/back/internal/pb/post.pb.go:882 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000110900, 0xb4f760, 0xc000111680, 0xc0002e0800, 0xc0000fb0e0, 0xf193a8, 0x0, 0x0, 0x0)
	/home/appins/go/pkg/mod/google.golang.org/grpc@v1.20.0/server.go:972 +0x486
google.golang.org/grpc.(*Server).handleStream(0xc000110900, 0xb4f760, 0xc000111680, 0xc0002e0800, 0x0)
	/home/appins/go/pkg/mod/google.golang.org/grpc@v1.20.0/server.go:1252 +0xd34
google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000025790, 0xc000110900, 0xb4f760, 0xc000111680, 0xc0002e0800)
	/home/appins/go/pkg/mod/google.golang.org/grpc@v1.20.0/server.go:691 +0xa5
created by google.golang.org/grpc.(*Server).serveStreams.func1
	/home/appins/go/pkg/mod/google.golang.org/grpc@v1.20.0/server.go:689 +0xa5
```

## Steps To Reproduce The Problem

  1. Start backend with  ./openbsrv --dbpass=$OBPASS --migrate
  2. Visit http://localhost:4243/v/posts 
  
## Specifications

  - Platform: Linux 5.8.0-44-generic
  - Browser: Firefox (unimportant)
  - Go version:  go1.15.8
  - Protoc version: libprotoc 3.15.3

## Additional Notes

I'd really like to learn and contribute to this project and would like to be the one to fix this issue, if that is possible.

--- a/back/internal/postsvc/internal/postdb/qrypost.go
+++ b/back/internal/postsvc/internal/postdb/qrypost.go
@@ -117,9 +117,14 @@ func (s *PostDB) findPosts(ctx cx, x *pb.FndPostsReq, y *pb.PostsResp) error {
 	}
 	defer selStmt.Close()
 
+	var like string
+	if len(x.Keywords) > 0 {
+		like = x.Keywords[0]
+	}
+
 	rows, err := selStmt.Query(
-		"%"+x.Keywords[0]+"%",
-		"%"+x.Keywords[0]+"%",
+		"%"+like+"%",
+		"%"+like+"%",
 	)
 	if err != nil {
 		return err
@@ -149,8 +154,8 @@ func (s *PostDB) findPosts(ctx cx, x *pb.FndPostsReq, y *pb.PostsResp) error {
 
 	err = s.db.QueryRow(
 		"SELECT COUNT(*) FROM post WHERE title LIKE ? OR body LIKE ?",
-		"%"+x.Keywords[0]+"%",
-		"%"+x.Keywords[0]+"%",
+		"%"+like+"%",
+		"%"+like+"%",
 	).Scan(&y.Total)
 	if err != nil {
 		return err

--- a/back/tests/openbsrv/postsvc_test.go
+++ b/back/tests/openbsrv/postsvc_test.go
@@ -22,6 +22,7 @@ func TestPostClientServices(t *testing.T) {
 	postClnt := pb.NewPostClient(conn)
 	t.Run("Add type", postSvcAddAndFndTypesFn(ctx, conn, postClnt))
 	t.Run("Add and find posts", postSvcAddAndFndPostsFn(ctx, conn, postClnt))
+	t.Run("Find all posts", postSvcFindAllPosts(ctx, conn, postClnt))
 	t.Run("Add, find and edit post", postSvcEdtPostFn(ctx, conn, postClnt))
 	t.Run("Find and delete post", postSvcDelPostFn(ctx, conn, postClnt))
 }
@@ -252,4 +253,53 @@ func postSvcFndPost(ctx context.Context, conn *grpc.ClientConn, clnt pb.PostClie
 		return postID.Posts[0].Id, nil
 	}
 	return "", nil
+}
+
+func postsContain(resp *pb.PostsResp, post *pb.PostResp) bool {
+	for _, j := range resp.Posts {
+		if j.Title == post.Title && j.Body == post.Body && j.TypeId == post.TypeId {
+			return true
+		}
+	}
+	return false
+}
+
+func postSvcFindAllPosts(ctx context.Context, conn *grpc.ClientConn, clnt pb.PostClient) func(*testing.T) {
+	return func(t *testing.T) {
+		tests := []struct {
+			fndReq *pb.FndPostsReq
+			want   *pb.PostsResp
+		}{
+			{
+				&pb.FndPostsReq{Keywords: []string{}},
+				&pb.PostsResp{
+					Posts: []*pb.PostResp{
+						{
+							Title:  "test title postA first",
+							Body:   "test body of first post",
+							TypeId: "2",
+						},
+						{
+							Title:  "test title postB second",
+							Body:   "test body of second postB",
+							TypeId: "3",
+						},
+					},
+				},
+			},
+		}
+
+		for _, tc := range tests {
+			resp, err := clnt.FndPosts(ctx, tc.fndReq)
+			if err != nil {
+				t.Errorf("unexpected error: %v", err)
+			}
+
+			for _, post := range tc.want.Posts {
+				if !postsContain(resp, post) {
+					t.Errorf("couldn't find post with title: %s", post.Title)
+				}
+			}
+		}
+	}
 }

